core/src/main/java/com/aem/demo/core/servlets/LikeDislikeServlet.java

```diff
@@ -0,0 +1,53 @@
+package com.aem.demo.core.servlets;
+
+import com.aem.demo.core.services.LikeDislikeService;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.servlet.Servlet;
+import java.io.IOException;
+
+@Slf4j
+@Component(service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "= Like and Dislike Servlet",
+                "sling.servlet.methods=" + HttpConstants.METHOD_POST,
+                "sling.servlet.paths=" + "/services/reaction",
+                "sling.servlet.extensions=" + "json"
```

Comment:
You don't need an extension here. You don't actually even use it when posting a request in `newscard.html` line 34 ))